icu4c/source/common/locresdata.cpp

```diff
@@ -145,6 +140,8 @@ uloc_getTableStringWithFallback(const char *path, const char *locale,
                 break;
             }
             /* succeeded in opening the fallback bundle .. continue and try to fetch the item */
+            ures_initStackObject(table.getAlias());
```

Comment:
What are these for? To clear the table and subTable before the next iteration?
If so, I think it would be much cleaner and clearer to declare them inside the loop, at the beginning, where the old code had the two ures_initStackObject() calls.

Reply:
Done.